scripts/evaluation.py

```diff
@@ -0,0 +1,239 @@
+            root_raw = await resp.json()
+            print()
+        root = NittaNode.from_dict(root_raw)  # type: ignore
+    return root
+
+
+@asynccontextmanager
+async def run_nitta(
+    example: Path,
+    nitta_exe_path: str = "stack exec nitta -- ",
+    nitta_args: str = "",
+    nitta_env: dict = {},
+    port: int = 8092,
+) -> AsyncGenerator[Tuple[asyncio.subprocess.Process, str], None]:
+    nitta_baseurl = f"http://localhost:{port}"
+
+    cmd = f"{nitta_exe_path} -p={port} {nitta_args} {example}"
+    print(cmd)
+    env = os.environ.copy()
+    env.update(nitta_env or {})
+
+    proc = None
+    try:
+        preexec_fn = None if os.name == "nt" else os.setsid  # see https://stackoverflow.com/a/4791612
+        proc = await asyncio.create_subprocess_shell(
+            cmd,
+            cwd=str("."),
+            stdout=sys.stdout,
+            stderr=sys.stderr,
+            shell=True,
+            preexec_fn=preexec_fn,
+            env=env,
+        )
+
+        await sleep(2)
+
+        yield proc, nitta_baseurl
+    finally:
+        if proc is not None and proc.returncode is None:
+            pid = proc.pid
+            pgid = os.getpgid(pid)
+            os.killpg(pgid, signal.SIGTERM)
+            await proc.wait()
+
+
+async def select_best_by_evaluator(session, evaluator, node, nitta_baseurl, counters, children_limit=None):
+    counters[evaluator.__name__] += 1
+
+    if node.is_leaf:
+        if not node.is_finish:
+            return None
+
+        return node
+
+    try:
+        await retrieve_subforest(node, session, nitta_baseurl)
+    except ServerDisconnectedError:
+        return None
+
+    children = node.children
+
+    if children_limit:
+        children = children[:children_limit]
+
+    children = [(evaluator(child), child) for child in node.children]
+    children.sort(key=lambda v: v[0], reverse=True)
+
+    while children:
+        next_best_child = children.pop(0)[1]
+        result = await select_best_by_evaluator(
+            session,
+            evaluator,
+            next_best_child,
+            nitta_baseurl,
+            counters,
+            children_limit,
+        )
+        if result is not None:
+            return result
+
+    return None
+
+
+def old_evaluator(node: NittaNode):
+    return node.score
+
+
+def reset_counters():
+    global counters
+    counters = defaultdict(lambda: 0)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("example_paths", type=str, nargs="+", help="Paths to the example files")
+    parser.add_argument(
+        "--nitta_args",
+        type=str,
+        default="",
+        help="Additional arguments for Nitta",
+    )
+    return parser.parse_args()
+
+
+async def main(args):
+    examples = args.example_paths
+
+    result_list = []  # List to hold all results
+
+    for example in examples:
+        example_path = Path(example)
+
+        reset_counters()
```

Comment:
From my point of view -- it is a bad practice. Better approach: to push all data in the list with all results. Global state -- can raise a lot of problem and in this case I don't understand why it is necessary.